### PR TITLE
Update log level based on central configuration and support "off" configuration

### LINKED ISSUE #970: [META 321] Add log_level spec
Implementing elastic/apm#332

Note that the implementation of this spec also includes adding the option to [Kibana](https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/common/agent_configuration/setting_definitions/general_settings.ts).

### PR DESCRIPTION
**Pull request by russcam**
This PR allows updating the log level of the configured `IApmLogger` at runtime, based on the log level received from central configuration, and also supports the "off" central configuration log level.

An `ILogLevelSwitchable` interface is introduced that can be implemented by an `IApmLogger` to allow the log level to be updated by central configuration.

`ConsoleLogger` implements `ILogLevelSwitchable` but `NetCoreLogger` does not. The ability to change the log level of `NetCoreLogger` at runtime is controlled by the provider used with Microsoft.Extensions.Logging.

Closes #970

### CODE REVIEW

**Comment by russcam** [comment]
jenkins please run the tests

File: test/Elastic.Apm.Tests/BackendCommTests/CentralConfig/CentralConfigFetcherTests.cs [hunk 1]
```diff
@@ -23,6 +35,101 @@ public class CentralConfigFetcherTests : LoggingTestBase
 	{
 		public CentralConfigFetcherTests(ITestOutputHelper xUnitOutputHelper) : base(xUnitOutputHelper) { }
 
+		[Fact]
+		public void Should_Not_Update_Logger_That_Is_ILogLevelSwitchable()
```
Comment (gregkalapos) [inline test/Elastic.Apm.Tests/BackendCommTests/CentralConfig/CentralConfigFetcherTests.cs:39]: Shouldn't this be called `Should_Update_Logger_That_Is_ILogLevelSwitchable`? (and not `Should_Not_Update...`)

**Review by gregkalapos — APPROVED** [review]
LGTM

File: test/Elastic.Apm.Tests/BackendCommTests/CentralConfig/CentralConfigFetcherTests.cs (hunk 1 above)
Comment (russcam) [reply test/Elastic.Apm.Tests/BackendCommTests/CentralConfig/CentralConfigFetcherTests.cs:39]: yep, it should, thanks! (must have introduced when applying consistent naming 😄 )

**Pull request merged**